plugins/wallabag.koplugin/main.lua

```diff
@@ -622,270 +692,457 @@ function Wallabag:download(article)
-        if self:callAPI("GET", item_url, nil, "", local_path) then
-            return downloaded
+        if self:callAPI("GET", item_url, nil, nil, local_path) then
+            return downloaded -- 3
         else
-            return failed
+            return failed     -- 1
         end
     end
-    return skipped
-end
-
--- method: (mandatory) GET, POST, DELETE, PATCH, etc...
--- apiurl: (mandatory) API call excluding the server path, or full URL to a file
--- headers: defaults to auth if given nil value, provide all headers necessary if in use
--- body: empty string if not needed
--- filepath: downloads the file if provided, returns JSON otherwise
--- @treturn result or (nil, "network_error") or (nil, "json_error")
--- or (nil, "http_error", code)
----- @todo separate call to internal API from the download on external server
-function Wallabag:callAPI(method, apiurl, headers, body, filepath, quiet)
+
+    return skipped -- 2
+end                -- Wallabag:downloadArticle
+
+--- Call the wallabag API.
+-- See https://app.wallabag.it/api/doc/ for methods and parameters.
+-- @param method GET, POST, DELETE, PATCH, etc…
+-- @param url URL endpoint on wallabag server without hostname, or full URL for external link
+-- @param[opt] headers Defaults to Authorization for API endpoints, none for external
+-- @param[opt] body Body to include in the request, if needed
+-- @param[opt] filepath Downloads the file if provided, returns JSON otherwise
+-- @param[opt=false] quiet
+-- @treturn bool Whether the request was successful
+-- @treturn string Error type if unsuccessful, filepath if success with path, JSON if without
+-- @treturn int HTTP response code if unsuccessful (e.g. 404, 503, …)
+function Wallabag:callAPI(method, url, headers, body, filepath, quiet)
+    quiet = quiet or false
+
     local sink = {}
-    local request = {}
+    local request = { method = method }
 
     -- Is it an API call, or a regular file direct download?
-    if apiurl:sub(1, 1) == "/" then
+    --- @todo Separate call to internal API from the download on external server
+    if url:sub(1, 1) == "/" then
         -- API call to our server, has the form "/random/api/call"
-        request.url = self.server_url .. apiurl
-        if headers == nil then
-            headers = {
-                ["Authorization"] = "Bearer " .. self.access_token,
-            }
-        end
-    else
-        -- regular url link to a foreign server
-        local file_url = apiurl
-        request.url = file_url
-        if headers == nil then
-            -- no need for a token here
-            headers = {}
-        end
+        request.url = self.server_url .. url
+        request.headers = headers or {
+            ["Authorization"] = "Bearer " .. self.access_token,
+        }
+    else -- Assume full URL (e.g. https://…)
+        request.url = url
+        request.headers = headers or {}
     end
 
-    request.method = method
-    if filepath ~= "" then
+    if filepath ~= nil then
```

Comment:
This looks like a very suspicious change. Something more along these lines wouldn't look suspicious:
```suggestion
    if filepath and filepath ~= "" then
```

Reply:
That's okay.
<!-- Reviewable comment -OFX8XUc4mdFI_p_Iyz-:b-mamtq -->
<!-- Sent from Reviewable.io -->

Reply:
Do you mean it's okay if I include these checks? Or it's okay to leave it as is?
<!-- Reviewable comment -OFesUsq1WEYISuxrOEJ:bl9ddpc -->
<!-- Sent from Reviewable.io -->

Reply:
Okay to leave it as.
<!-- Reviewable comment -OFfjMWY80gVz8TemuLk:b-k0v2qa -->
<!-- Sent from Reviewable.io -->

Reply:
Done.
<!-- Reviewable comment -OFvAbd6-2GCMoQ7ZUXT:b-896fix -->
<!-- Sent from Reviewable.io -->